faucet/valve.py

```diff
@@ -1243,9 +1255,9 @@ def learn_host(self, now, pkt_meta, other_valves):
                 now, learn_port, pkt_meta.vlan, pkt_meta.eth_src,
                 last_dp_coldstart_time=self.dp.dyn_last_coldstart_time)
             if update_cache:
-                pkt_meta.vlan.add_cache_host(pkt_meta.eth_src, learn_port, now)
                 if pkt_meta.l3_pkt is None:
                     pkt_meta.reparse_ip()
+                pkt_meta.vlan.add_cache_host(pkt_meta.eth_src, learn_port, now, str(pkt_meta.l3_src))
```

Comment:
why str()?   When is this ever not None or a string already?

Reply:
Just following existing code. Before raising the event 'L2_LEARN' below, it is cast as string. Assumed there must be reason for it.

Reply:
Safe to leave it as it is?

Reply:
If that's the case, then the str() cast should be where it constructs the event, not here.